streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/ThreadMetrics.java

```diff
@@ -220,6 +231,70 @@ public static Sensor commitOverTasksSensor(final String threadId,
         return commitOverTasksSensor;
     }
 
+    public static Sensor processRatioSensor(final String threadId,
```

Comment:
Could you please add unit tests for the new sensors and metrics?

Reply:
Yes, they are added in the follow-up PR.